### PR TITLE
support k8s-cluster in init_orca_context, init_nncontext

### PR DESCRIPTION
**Pull request by Le-Zheng**
k8s cluster `spark-submit`:
```
/opt/spark/bin/spark-submit \
  --master k8s://https://127.0.0.1:8443 \
  --deploy-mode cluster \
  --conf spark.kubernetes.authenticate.driver.serviceAccountName=spark \
  --name bigdl2-pytorch_estimator \
  --conf spark.kubernetes.container.image=10.239.45.10/arda/intelanalytics/bigdl-k8s-spark-3.1.2:0.14.0-SNAPSHOT \
  --conf spark.executor.instances=1 \
  --conf spark.kubernetes.driver.volumes.persistentVolumeClaim.nfsvolumeclaim.options.claimName=nfsvolumeclaim \
  --conf spark.kubernetes.driver.volumes.persistentVolumeClaim.nfsvolumeclaim.mount.path=/bigdl2.0/data \
  --conf spark.kubernetes.executor.volumes.persistentVolumeClaim.nfsvolumeclaim.options.claimName=nfsvolumeclaim \
  --conf spark.kubernetes.executor.volumes.persistentVolumeClaim.nfsvolumeclaim.mount.path=/bigdl2.0/data \
  --conf spark.kubernetes.driver.label.az=true \
  --conf spark.kubernetes.executor.label.az=true \
  --conf spark.kubernetes.node.selector.spark=true \
  --conf spark.kubernetes.driverEnv.http_proxy=http://child-prc.intel.com:913 \
  --conf spark.kubernetes.driverEnv.https_proxy=http://child-prc.intel.com:913 \
  --conf spark.kubernetes.executorEnv.http_proxy=http://child-prc.intel.com:913 \
  --conf spark.kubernetes.executorEnv.https_proxy=http://child-prc.intel.com:913 \
  --conf spark.kubernetes.container.image.pullPolicy=Always \
  --conf spark.pyspark.driver.python=./env/bin/python \
  --conf spark.pyspark.python=./env/bin/python \
  --archives http://10.239.45.10:8081/repository/raw/spark-k8s-remote-storage/environment_test.tar.gz#env \
  --conf spark.kubernetes.executor.deleteOnTermination=false \
  --executor-cores 16 \
  --executor-memory 50g \
  --total-executor-cores 64 \
  --driver-cores 4 \
  --driver-memory 50g \
  --properties-file /opt/bigdl-0.14.0-SNAPSHOT/conf/spark-bigdl.conf \
  --py-files local:///opt/bigdl-0.14.0-SNAPSHOT/python/bigdl-spark_3.1.2-0.14.0-SNAPSHOT-python-api.zip,local:///opt/bigdl-0.14.0-SNAPSHOT/examples/orca/learn/horovod/pytorch_estimator.py \
  --conf spark.driver.extraJavaOptions=-Dderby.stream.error.file=/tmp \
  --conf spark.sql.catalogImplementation='in-memory' \
  --conf spark.driver.extraClassPath=local:///opt/bigdl-0.14.0-SNAPSHOT/jars/* \
  --conf spark.executor.extraClassPath=local:///opt/bigdl-0.14.0-SNAPSHOT/jars/* \
  local:///opt/bigdl-0.14.0-SNAPSHOT/examples/orca/learn/horovod/pytorch_estimator.py --cluster_mode 'spark-submit'
```

### CODE REVIEW

File: python/dllib/src/bigdl/dllib/utils/spark.py [hunk 1]
```diff
@@ -363,6 +363,61 @@ def init_spark_on_k8s(self,
...
+                     "spark.kubernetes.container.image": container_image})
+        if "BIGDL_CLASSPATH" in os.environ:
+            zoo_bigdl_jar_path = os.environ["BIGDL_CLASSPATH"]
+        else:
+            zoo_bigdl_jar_path = get_zoo_bigdl_classpath_on_driver()
+        if "spark.executor.extraClassPath" in conf:
+            conf["spark.executor.extraClassPath"] = "{}:{}".format(
+                zoo_bigdl_jar_path, conf["spark.executor.extraClassPath"])
+        else:
+            conf["spark.executor.extraClassPath"] = zoo_bigdl_jar_path
```
Comment (hkvision) [inline python/dllib/src/bigdl/dllib/utils/spark.py:416]: Can we avoid duplicating these code for client and cluster?
Also the driver host and port are missing?

**Comment by hkvision** [comment]
Is this working?
Yarn cluster seems needs to launch a separate process: https://github.com/intel-analytics/BigDL/blob/branch-2.0/python/dllib/src/bigdl/dllib/utils/spark.py#L210

**Comment by Le-Zheng** [comment]
> Is this working? Yarn cluster seems needs to launch a separate process: https://github.com/intel-analytics/BigDL/blob/branch-2.0/python/dllib/src/bigdl/dllib/utils/spark.py#L210

I am not sure. Let me test and verify.

File: python/dllib/src/bigdl/dllib/utils/spark.py (hunk 1 above)
Comment (Le-Zheng) [reply python/dllib/src/bigdl/dllib/utils/spark.py]: cluster mode do not need that.

**Comment by Le-Zheng** [comment]
updated according to comments above and move to https://github.com/intel-analytics/BigDL/pull/4008

**Pull request closed without merging**